modules/aggregation-layers/src/new-grid-layer/new-grid-layer.js

```diff
@@ -0,0 +1,155 @@
+import {CompositeLayer} from '@deck.gl/core';
+import {AGGREGATION_OPERATION} from '../utils/gpu-grid-aggregation/gpu-grid-aggregator-constants';
+import GPUGridAggregator from '../utils/gpu-grid-aggregation/gpu-grid-aggregator';
+import GPUGridLayer from '../gpu-grid-layer/gpu-grid-layer';
+import GridLayer from '../grid-layer/grid-layer';
+
+const defaultProps = Object.assign({}, GPUGridLayer.defaultProps, GridLayer.defaultProps);
+
+// Function to convert from getWeight/aggregation props to getValue prop for Color and Elevation
+function getMean(pts, accessor) {
+  const filtered = pts.map(item => accessor(item)).filter(pt => Number.isFinite(pt));
+
+  return filtered.length ? filtered.reduce((accu, curr) => accu + curr, 0) / filtered.length : null;
```

Comment:
Perf: avoid defining inline function in frequently used utility functions.

```js
const filtered = pts.map(accessor).filter(Number);
filtered.reduce(sumReducer, 0);
```

Reply:
Addressed.